Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4799,22 +4827,22 @@ - (void)reloadViewItems
                 // ...and always show the "disappearing messages" animation.
                 shouldHideFooter
                     = ([timestampText isEqualToString:nextTimestampText] && receiptStatus == nextReceiptStatus
-                        && outgoingMessage.messageState != TSOutgoingMessageStateFailed && !nextViewItem.shouldShowDate
-                        && !isDisappearingMessage);
+                        && outgoingMessage.messageState != TSOutgoingMessageStateFailed
+                        && !nextViewItem.hasDateOrUnreadBreak && !isDisappearingMessage);
```

Comment:
Issue 2: Don't cluster messages across unread indicator breaks.

Reply:
TIOLI, but this might be more legible as a closure with a series of guards.

```
shouldHideFooter = ^{

    if (![timestampText isEqualToString:nextTimestampText]) {
      return NO;
    }

    if (recipientStatus != nextRecipientStatus) {
      return NO;
    }

    ...

    return YES;
};
```

Or a series of chained else ifs...